src/libstd/macros.rs

```diff
@@ -124,6 +130,45 @@ macro_rules! println {
     ($fmt:expr, $($arg:tt)*) => (print!(concat!($fmt, "\n"), $($arg)*));
 }
 
+/// Macro for printing to the standard error.
+///
+/// Equivalent to the `print!` macro, except that output goes to
+/// `io::stderr` instead of `io::stdout`.  See `print!` for
+/// example usage.
+///
+/// Use `eprint!` only for error and progress messages.  Use `print!`
+/// instead for the primary output of your program.
+///
+/// # Panics
+///
+/// Panics if writing to `io::stderr` fails.
+#[macro_export]
+#[unstable(feature = "eprint", issue="40528")]
+#[allow_internal_unstable]
+macro_rules! eprint {
+    ($($arg:tt)*) => ($crate::io::_eprint(format_args!($($arg)*)));
+}
+
+/// Macro for printing to the standard error, with a newline.
+///
+/// Equivalent to the `println!` macro, except that output goes to
+/// `io::stderr` instead of `io::stdout`.  See `println!` for
+/// example usage.
+///
+/// Use `eprintln!` only for error and progress messages.  Use `println!`
+/// instead for the primary output of your program.
+///
+/// # Panics
+///
+/// Panics if writing to `io::stderr` fails.
+#[macro_export]
+#[unstable(feature = "eprint", issue="40528")]
+macro_rules! eprintln {
+    () => (eprint!("\n"));
+    ($fmt:expr) => (eprint!(concat!($fmt, "\n")));
+    ($fmt:expr, $($arg:tt)*) => (eprint!(concat!($fmt, "\n"), $($arg)*));
```

Comment:
@ollie27's suggested change to the macro does work, but I am surprised that it works - it appears to be a type error.  I am reluctant to do something that may only work by accident.  It also appears to have double formatting overhead, which is not desirable.  (It is disappointing that macro_rules! macros have no direct way to specify that a particular token tree should be a string literal.  Freaking CPP can do this.)

But more importantly, I really do think `eprintln` should be bug-for-bug compatible with `println`, because people should be able to go through existing programs and change `println` to `eprintln` where appropriate, _without thinking about it_ apart from "should this properly go to stderr?"  Again, if this is to be changed, it should be changed for both `println` and `eprintln` at the same time.